chevah/brink
# Add support to run GitHub Action tests without a commit.

Scope
=====

This add support to run tests on Github Actions without a commit.

It updates the pavement helpers and updates the GitHub actions configration.


Changes
=======

Update the workflow to handle manual runs...and job and test selection.

The try.yaml is there just for experiments.


How to try and test the changes
===============================

reviewers: @dumol 

Give it a try.

There are some differences from Buildbot.

We don't have `builders` but `workflows` and while a `builder` contains a single job on a single OS, a workflow can contain test runs on multiple OS (on a build matrix).

So a "workflow" is somehow similar to Buildbot projects and  a "job" is similar to a Buildbot builder. 
And both GitHub Actions and Buildbot have steps.

You run a specific workflow. This is required. 
You can optionally run only a specific job inside the workfow.
You can optionally run only a specific test suite inside the job or inside all the jobs.
You can optionally only show output a specific step inside the job or inside all the jobs.


```
$ paver actions_try --workflow=main.yml  --job=linux_unicode_path --step=test --test=test_filesystem
```

----------

Note that you can now also do a manual run of a test from the UI if you go at https://github.com/chevah/brink/actions?query=workflow%3ATry-Patch

For the GitHub-CI you will see the GitHub UI once this is merged in master
